File: src/org/lasalle/clima/excel/bottomuphour/scripting/BottomUpAnalysis.js
/**
 * @object BottomUp
 * Bottom up analysis.
 */
var BottomUp = { 
	/**
	* @function process(row, value)
	* This function process the given input files to produce the result in the output folder. Each
	* output file corresponds to a pollutant in an specific hour.
	* @param file1:String The path to the excel file containing the information about length and flows on each cell of map.
	* @param file2:String The path to the excel file containing the information about vehicular flows by hour.
	* @param file3:String The path to the excel file containing the information about pollutant emission factors.
	* @param outputFolder:String The path to store the excel outputs.
	* @param types:Object An array containing the type of days to process. "H" for habil, "D" for domingo and "S" for sabado.
	*/
	process : function(file1, file2, file3, outputFolder, types){
		 file1 = File.checkGlobal(file1);
		 file2 = File.checkGlobal(file2);
		 file3 = File.checkGlobal(file3);
		 outputFolder = File.checkGlobal(outputFolder);
		  
		 BottomUpAPI.process(file1, file2, file3, outputFolder, types, Global.basedir, Global.basedirkey);
	}
}

File: src/org/lasalle/clima/excel/bottomuphour/scripting/BottomUpAnalysis.java
package org.lasalle.clima.excel.bottomuphour.scripting;

import java.io.File;

import org.cheminfo.function.Function;
import org.cheminfo.function.scripting.SecureFileManager;
import org.lasalle.clima.excel.bottomuphour.Aggregator;
/**
 * 
 * @author acastillo
 *
 */

public class BottomUpAnalysis extends Function{
	
	public boolean process(String file1, String file2, String file3, String outputFolder, String[] types, String basedir, String basedirkey){
		//this.appendError("BottomUp::process", file1);
		//To create the output folder if different than the input folder
		String file1Full = SecureFileManager.getValidatedFilename(
				basedir, basedirkey, file1);
		if (file1Full == null){
			this.appendError("BottomUpAnalysis::process","Could not get the path to "+file1);
			return false;
		}
		String file2Full = SecureFileManager.getValidatedFilename(
				basedir, basedirkey, file2);
		if (file2Full == null){
			this.appendError("BottomUpAnalysis::process","Could not get the path to "+file2);
			return false;
		}
		String file3Full = SecureFileManager.getValidatedFilename(
				basedir, basedirkey, file3);
		if (file3Full == null){
			this.appendError("BottomUpAnalysis::process","Could not get the path to "+file3);
			return false;
		}
		String outputFolderFull = SecureFileManager.getValidatedFilename(
				basedir, basedirkey, outputFolder);
		if (outputFolderFull == null){
			this.appendError("BottomUpAnalysis::process","Could not get the path to "+outputFolder);
			return false;
		}
		File output = new File(outputFolderFull);
		if(!output.exists()){
			if(output.mkdirs())
				this.appendInfo("BottomUpAnalysis::process","Output folder created: "+outputFolder);
		}
		
		Aggregator processor = new Aggregator(new File(file1Full),
				new File(file2Full), 
				new File(file3Full),
				new File(outputFolderFull),this);
		processor.setTypes(types);
		processor.expandCells();
		return true;
	}
}
